src/id/doujindesu/src/eu/kanade/tachiyomi/extension/id/doujindesu/DoujinDesu.kt

```diff
@@ -0,0 +1,320 @@
+        Genre("Glasses", "glasses"),
+        Genre("Gore", "gore"),
+        Genre("Group", "group"),
+        Genre("Guro", "guro"),
+        Genre("Gyaru", "gyaru"),
+        Genre("Hairy", "hairy"),
+        Genre("Handjob", "handjob"),
+        Genre("Harem", "harem"),
+        Genre("Horns", "horns"),
+        Genre("Huge Breast", "huge-breast"),
+        Genre("Humiliation", "humiliation"),
+        Genre("Impregnation", "impregnation"),
+        Genre("Incest", "incest"),
+        Genre("Inflation", "inflation"),
+        Genre("Insect", "insect"),
+        Genre("Inseki", "inseki"),
+        Genre("Inverted Nipples", "inverted-nipples"),
+        Genre("Invisible", "invisible"),
+        Genre("Kemomimi", "kemomimi"),
+        Genre("Kimono", "kimono"),
+        Genre("Lactation", "lactation"),
+        Genre("Leotard", "leotard"),
+        Genre("Lingerie", "lingerie"),
+        Genre("Loli", "loli"),
+        Genre("Lolipai", "lolipai"),
+        Genre("Maid", "maid"),
+        Genre("Males Only", "males-only"),
+        Genre("Masturbation", "masturbation"),
+        Genre("Miko", "miko"),
+        Genre("MILF", "milf"),
+        Genre("Mind Break", "mind-break"),
+        Genre("Mind Control", "mind-control"),
+        Genre("Minigirl", "minigirl"),
+        Genre("Miniguy", "miniguy"),
+        Genre("Monster", "monster"),
+        Genre("Monster Girl", "monster-girl"),
+        Genre("Mother", "mother"),
+        Genre("Multi-work Series", "multi-work-series"),
+        Genre("Muscle", "muscle")
+    )
+
+    private val categoryNames = arrayOf(
+        Category("All", ""),
+        Category("Manga", "Manga"),
+        Category("Manhua", "Manhua"),
+        Category("Doujinshi", "Doujinshi"),
+        Category("Manhwa", "Manhwa")
+    )
+
+    private class CategoryNames(categories: Array<Category>) : Filter.Select<Category>("Category", categories, 0)
+    private class OrderBy(orders: Array<Order>) : Filter.Select<Order>("Order", orders, 0)
+    private class GenreList(genres: Array<Genre>) : Filter.Select<Genre>("Genre", genres, 0)
+    private class StatusList(statuses: Array<Status>) : Filter.Select<Status>("Status", statuses, 0)
+
+    private fun basicInformationFromElement(element: Element): SManga {
+        val manga = SManga.create()
+
+        manga.title = element.select("div > div > a").attr("alt")
+        manga.setUrlWithoutDomain(element.select("div > div > a").attr("href"))
+        manga.thumbnail_url = element.select("div > div > a > div > img").attr("src")
+
+        return manga
+    }
+
+    private fun getNumberFromString(epsStr: String): String {
+        return epsStr.filter { it.isDigit() }
+    }
+
+    private fun reconstructDate(Str: String): Long {
+        val pattern = SimpleDateFormat("MMMM d, yyyy", Locale("id"))
+        return when {
+            pattern.parse(Str) != null -> pattern.parse(Str).time
+            else -> System.currentTimeMillis()
+        }
+    }
```

Comment:
If you return the current time if the parsing fails, it will always change and the chapter will be detected as a new one.

```suggestion
    private fun reconstructDate(dateStr: String): Long {
        return runCatching { DATE_FORMAT.parse(dateStr)?.time }
            .getOrNull() ?: 0L
    }
```

And then make the `DATE_FORMAT` a constant, otherwise it will be recreated every chapter:

```kotlin
companion object {
    private val DATE_FORMAT by lazy {
        SimpleDateFormat("MMMM d, yyyy", Locale("id"))
    }
}
```